File: cpp/1st_Week/Praktikum/PersegiPanjang.cpp
#include <iostream>
#include <string>

class PersegiPanjang
{
    private:
        //atribut
        int panjang;
        int lebar;
    public:
        //constructor
        PersegiPanjang(){
            this->panjang = this->lebar = 0;
        }

        PersegiPanjang(int p, int l){
            this->lebar = l;
            this->panjang = p;
        }

        //method getter setter
        void setPanjang(int panjang){
            this->panjang = panjang;
        }

        void setLebar(int lebar){
            this->lebar = lebar;
        }

        int getPanjang(){
            return this->panjang;
        }

        int getLebar(){
            return this->lebar;
        }

        int keliling(){
            return 2*(this->panjang+this->lebar);
        }

        int luas(){
            return this->panjang * this->lebar;
        }

        //destructor
        ~PersegiPanjang(){}

};

File: python/1st_week/Praktikum/PersegiPanjang.py
class PersegiPanjang():
    # private atribut
    __panjang = 0
    __lebar = 0
    
    def __init__(self, panjang = 0, lebar = 0):
        self.__panjang = panjang
        self.__lebar = lebar

    def setPanjang(self, panjang):
        self.__panjang = panjang

    def setLebar(self, lebar):
        self.__lebar = lebar

    def getPanjang(self):
        return self.__panjang

    def getLebar(self):
        return self.__lebar
    
    def keliling(self):
        return 2 * (self.__panjang + self.__lebar)
    
    def luas(self):
        return self.__panjang * self.__lebar
